### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by yegonkimutai**
In this PR, my partner and I did the following ;
- [ ] Solved the N + 1 problem while fetching posts
  Before:
Screenshot (87)

  After:
Screenshot (88)

- [ ] Use Capybara to write integration tests.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ Metrics/ClassLength:
   Max: 150
 Metrics/BlockLength:
   AllowedMethods: ['describe']
-  Max: 30
+  Max: 50
```
Comment (Ismailco) [inline .rubocop.yml]: - [ ] Kindly don't change the robocop rules, instead try to reduce the size of your methods to reflect the rules

File: spec/integration/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  scenario 'See the number of posts the user has written' do
+    expect(page).to have_content(@user.posts.count)
+  end
+
+  scenario 'See the bio of the user' do
+    expect(page).to have_content(@user.bio)
+  end
+
+  scenario "I can see a button that lets me view all of a user's posts" do
```
Comment (Ismailco) [inline spec/integration/index_spec.rb]: - [ ] Kindly make sure to add tests for these functionalities:
   - * Tests that redirect to the post's show page when you click on a user's post.
   - * Tests that redirect to the user's post's index page when you click on a **see all posts**.

File: spec/integration/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    expect(page).to have_link('See all posts', href: user_posts_path(@user))
+  end
```
Comment (Ismailco) [inline spec/integration/index_spec.rb]: - [ ] Kindly consider moving the tests for the show page to a separate file instead

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @yegonkimutai, @Bestbynature, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @yegonkimutai,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**